lib/Cro/TCP/NoDelay.pm6

```diff
@@ -0,0 +1,25 @@
+unit module Cro::TCP::NoDelay;
+
+use NativeCall;
+
+# Reduce per-socket cost for checking OS
+INIT my $is-win = $*DISTRO.is-win;
+
+constant TCP_NODELAY = 1;
+
+sub setsockopt(int32 $sockfd, int32 $level, int32 $optname,
+               CArray[int32] $optval, int32 $optlen --> int32) is native { * }
+
+sub nodelay($socket) is export {
+    # Not tested on Windows systems, so silently do nothing there
+    return if $is-win;
+
+    my $nd   = $socket.native-descriptor;
+    my $on   = CArray[int32].new(1);
+    my $size = nativesizeof(int32) * $on.elems;
+
+    if setsockopt($nd, PROTO_TCP, TCP_NODELAY, $on, $size) {
```

Comment:
Where does `PROTO_TCP` come from?

Reply:
Ah, from `CORE.setting`. Alright. Wonder how portable those constants are...

Reply:
(Of course, putting it in here won't make it more portable, so it's the right place to take it from. :-))